packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -140,6 +140,15 @@ library LibProposing {
         emit BlockProposed(state.nextBlockId++, meta);
     }
 
+    function getProposedBlock(
+        TaikoData.State storage state,
+        uint256 maxNumBlocks,
+        uint256 id
+    ) internal view returns (TaikoData.ProposedBlock storage) {
+        require(id > state.latestVerifiedId && id < state.nextBlockId, "L1:id");
+        return LibUtils.getProposedBlock(state, maxNumBlocks, id);
```

Comment:
do we need to keep using the `state.getProposedBlock(maxNumBlocks, id);` style here? Like the code in `LibVerifying.sol` and `LibProving.sol`.

Reply:
changed